src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs

```diff
@@ -9,6 +10,28 @@ namespace Avalonia.Threading
     /// </summary>
     public class AvaloniaSynchronizationContext : SynchronizationContext
     {
+        internal readonly DispatcherPriority Priority;
+        private readonly NonPumpingLockHelper.IHelperImpl? _nonPumpingHelper =
+            AvaloniaLocator.Current.GetService<NonPumpingLockHelper.IHelperImpl>();
+        
+        public AvaloniaSynchronizationContext():  this(Thread.CurrentThread.GetApartmentState() == ApartmentState.STA)
```

Comment:
I wonder. It is possible to set apartment state on the thread on non-windows platforms (in fact, we set it in our templates). Won't it cause any unnecessary operations here on non-windows?

Reply:
Ok, it seems like IHelperImpl is registered on windows only. And tests.